src/charm.py

```diff
@@ -318,20 +349,35 @@ def _ingress_config(self) -> dict:
                 # see https://doc.traefik.io/traefik/v2.0/user-guides/grpc/#with-http-h2c
                 http_services[
                     f"juju-{self.model.name}-{self.model.app.name}-service-{sanitized_protocol}"
-                ] = {"loadBalancer": {"servers": [{"url": f"h2c://{self.hostname}:{port}"}]}}
+                ] = {"loadBalancer": {"servers": self.generate_lb_server_config("h2c", port)}}
             else:
                 # anything else, including secured GRPC, can use _internal_url
                 # ref https://doc.traefik.io/traefik/v2.0/user-guides/grpc/#with-https
                 http_services[
                     f"juju-{self.model.name}-{self.model.app.name}-service-{sanitized_protocol}"
-                ] = {"loadBalancer": {"servers": [{"url": f"{self._internal_url}:{port}"}]}}
+                ] = {
+                    "loadBalancer": {"servers": self.generate_lb_server_config(self._scheme, port)}
+                }
         return {
             "http": {
                 "routers": http_routers,
                 "services": http_services,
             },
         }
 
+    def generate_lb_server_config(self, scheme: str, port: int) -> List[Dict[str, str]]:
```

Comment:
```suggestion
    def _build_lb_server_config(self, scheme: str, port: int) -> List[Dict[str, str]]:
```

for consistency with the other `_build_*_config` methods